### PR TITLE
Fix pull dataset by account

### PR DESCRIPTION
**Pull request by rmn-boiko**
## Description

Closes: [Pulling with --all flag in a multi-tenant workspace should only match current account datasets](https://github.com/kamu-data/kamu-cli/issues/532)

## Checklist before requesting a review

- [x] [CHANGELOG.md](./CHANGELOG.md) updated
- [x] API changes are backwards-compatible
- [x] Workspace layout changes include a migration
- [x] Documentation update PR: <link or N/A>
- [x] Dataset pipelines update scheduled if needed
- [x] Unit-tests added

### CODE REVIEW

File: src/app/cli/src/commands/pull_command.rs [hunk 1]
```diff
@@ -104,11 +107,14 @@ impl PullCommand {
...
         &self,
         listener: Option<Arc<dyn PullMultiListener>>,
+        current_account_name: Option<AccountName>,
     ) -> Result<Vec<PullResponse>, CLIError> {
         let dataset_refs: Vec<_> = filter_datasets_by_any_pattern(
             self.dataset_repo.as_ref(),
             self.search_svc.clone(),
             self.refs.clone(),
+            current_account_name,
+            self.all,
```
Comment (zaychenko-sergei) [inline src/app/cli/src/commands/pull_command.rs:117]: I don't think that switch `-all` and list of references are a valid combination. If I use `-all`, I would expect to update all of my datasets without specifying any patterns/refs. If that is so, then it's quite weird to mix extraction of all account datasets and "-all" switch in 1 filtering algorithm.

Comment (rmn-boiko) [reply src/app/cli/src/commands/pull_command.rs:117]: We will not allow the use of both refs and `--all`. But as we are encapsulating a logic of fetching datasets to process, I decided it's a good place to handle this flag. And one more reason that `--all` will not mean like really all records in the case of multitenant workspace.

Comment (rmn-boiko) [reply src/app/cli/src/commands/pull_command.rs:117]: Agreed to bring changes and move `-all` flag handling back to pull implementation level

File: src/domain/core/src/services/pull_service.rs [hunk 2]
```diff
@@ -124,6 +125,8 @@ pub struct PullMultiOptions {
     pub recursive: bool,
     /// Pull all known datasets
     pub all: bool,
+    /// Owner of datasets
+    pub current_account_name: AccountName,
```
Comment (zaychenko-sergei) [inline src/domain/core/src/services/pull_service.rs:129]: Adding this parameter is unnecessary.
`PullServiceImpl` already injects `CurrentAccountSubject`, so no need to propagate this value.

**Pull request merged**